source-control/git-guidance/readme.md

```diff
@@ -0,0 +1,168 @@
+# Git Guidance
+
+## The basics
+
+### Installation
+
+[Install Git](https://git-scm.com/downloads) and follow the [First-Time Git Setup](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup).
+
+### Cloning
+
+<!-- TODO: add more information about cloning repos (e.g., SSH, https, etc.) -->
+
+### Committing
+
+<!-- TODO: add more information about commiting changes (e.g., commit messages, setting editors, etc.) -->
+
+### Merging
+
+<!-- TODO: add more information about merging changes (e.g., links) -->
+
+### Managing remotes
+
+<!-- TODO: add more information about managing remotes (e.g., origin, other GitHub forks, etc., SSH vs. HTTPS) -->
+
+## Rolling back changes
+
+### Reverting and deleting commits
+
+To "undo" a commit, run the following two commands: `git revert` and `git reset`. `git revert` creates a new commit that undoes commits while `git reset` allows to delete commits entirely from the commit history.
+
+> If you have committed secrets/keys, `git reset` will remove them from the commit history!
+
+To **delete** the latest commit use `HEAD~`:
+
+```bash
+git reset --hard HEAD~1
+```
+
+To delete commits back to a specific commit, use the respective commit id:
+
+```bash
+git reset --hard <sha1-commit-id>
+```
+
+after you deleted the unwanted commits, push using `force`:
+
+```bash
+git push origin HEAD --force
+```
+
+<!-- TODO: add information about interactive rebase -->
+
+### Using the reflog
+
+<!-- TODO: add information about using the reflog to recover "lost" changes -->
+
+## Using submodules
+
+Submodules can be useful in more complex deployment and/or development scenarios
+
+Adding a submodule to your repo
+
+```bash
+git submodule add -b master <your_submodule>
+```
+
+Initialize and pull a repo with submodules:
+
+```bash
+git submodule init
+git submodule update --init --remote
+git submodule foreach git checkout master
+git submodule foreach git pull origin
+```
+
+## Stashing changes
```

Comment:
Note to other reviewers as it is hard to spot this change... this was changed from **Stashing** to **Stashing changes**
I'm good with either - but since you changed from Naming branches to Branch naming this is now a bit inconsistent

Reply:
True.